mesop/component_helpers/helper.py

```diff
@@ -374,14 +380,63 @@ def register_event_handler(
   return fn_id
 
 
-@lru_cache(maxsize=None)
+def has_stable_repr(obj: Any) -> bool:
+  """Check if an object has a stable repr."""
```

Comment:
[nit] Maybe add a bit more detail in this comment. Specifically would be good to know the potential limitations of using partial functions.